From ca55305e9c58aa7c1f99de8fb0f6195f48891029 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Mon, 26 May 2025 15:30:19 +0800
Subject: [PATCH 009/119] mtk: mt76: mt7996: fix unified command header

For the SDO, driver does not need to wait ACK event so do not mark MCU_CMD_ACK.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/mcu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 195c8f2c..09c9627d 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -294,6 +294,9 @@ mt7996_mcu_send_message(struct mt76_dev *mdev, struct sk_buff *skb,
 		else
 			uni_txd->option = MCU_CMD_UNI_EXT_ACK;
 
+		if (mcu_cmd == MCU_UNI_CMD_SDO)
+			uni_txd->option &= ~MCU_CMD_ACK;
+
 		if ((cmd & __MCU_CMD_FIELD_WA) && (cmd & __MCU_CMD_FIELD_WM))
 			uni_txd->s2d_index = MCU_S2D_H2CN;
 		else if (cmd & __MCU_CMD_FIELD_WA)
-- 
2.45.2

